src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java

```diff
@@ -36,52 +59,183 @@ public void shouldGetValidMetaDataResponse() {
+            () -> assertNotNull(response.getBody()),
+            () -> assertThat(
+                response.getBody(),
+                is("Document doesn't exist for requested documetd id at CCD Data Store API Side")
+            )
+        );
+    }
 
-        ResponseEntity<Object> response = testee.getDocumentBinaryContentbyDocumentId(serviceAuthorization, getUuid(), "", "");
-        assertNotNull(response, "Invalid Response from API");
-        assertNotNull(response.getBody(), "Empty response body");
-        assertEquals(HttpStatus.OK, response.getStatusCode(), "Status code is OK");
+    @Test
+    @DisplayName("should throw 403 forbidden when the requested document does not match with available doc")
+    public void shouldThrowForbiddenWhenDocumentDoesNotMatch() {
+        CaseDocumentMetadata caseDocumentMetadata = getCaseDocumentMetadata(
+            UNMATCHED_DOCUMENT_ID,
+            Arrays.asList(
+                Permission.CREATE,
+                Permission.READ
+            )
+        );
+        doReturn(setDocumentMetaData()).when(documentManagementService).getDocumentMetadata(getUuid());
+        doReturn(CASE_ID).when(documentManagementService).extractCaseIdFromMetadata(setDocumentMetaData().getBody());
+        doReturn(caseDocumentMetadata).when(caseDataStoreService).getCaseDocumentMetadata(CASE_ID, getUuid());
+        doReturn(setDocumentBinaryContent("FORBIDDEN")).when(documentManagementService).getDocumentBinaryContent(getUuid());
+
+        ResponseEntity<Object> response = testee.getDocumentBinaryContentbyDocumentId(
+            serviceAuthorization,
+            getUuid(),
+            "",
+            ""
+        );
+
+        assertAll(
+            () -> verify(documentManagementService).getDocumentMetadata(getUuid()),
+            () -> assertThat(response.getStatusCode(), is(HttpStatus.FORBIDDEN)),
+            () -> assertNotNull(response.getBody()),
+            () -> assertThat(response.getBody(), is("User don't have read permission on requested document"))
+        );
     }
 
     private ResponseEntity<StoredDocumentHalResource> setDocumentMetaData() {
         StoredDocumentHalResource resource = new StoredDocumentHalResource();
         resource.setCreatedBy("test");
         resource.setOriginalDocumentName("test.png");
-        return new ResponseEntity<StoredDocumentHalResource>(resource, HttpStatus.ACCEPTED);
+        return new ResponseEntity<StoredDocumentHalResource>(resource, HttpStatus.OK);
+    }
+
+    private UUID getUuid() {
+        return UUID.fromString(MATCHED_DOCUMENT_ID);
     }
 
-    private ResponseEntity<Object> setDocumentBinaryContent() {
-        return new ResponseEntity<Object>(new Object(), HttpStatus.ACCEPTED);
+    private CaseDocumentMetadata getCaseDocumentMetadata(String docId, List<Permission> permission) {
+        Document document = Document.builder().permissions(permission).build();
+        document.setId(docId);
```

Comment:
remove this line and set docId from builder.

Reply:
Done